src/black/__init__.py

```diff
@@ -613,7 +626,13 @@ def get_sources(
     gitignore = get_gitignore(root)
 
     for s in src:
-        p = Path(s)
+        if s == "-" and stdin_filename:
+            p = Path(stdin_filename)
+            is_stdin = True
+        else:
+            p = Path(s)
+            is_stdin = False
+
         if p.is_dir():
```

Comment:
```suggestion
        if p.is_dir() and not is_stdin:
```

If `--stdin-filename` just happens to point to a directory that exists, Black wrongly switches to in-place file formatting. 
```console
(black) ichard26@acer-ubuntu:~/programming/oss/black$ echo "eggs" | black --stdin-filename="../black" -
All done! ✨ 🍰 ✨
15 files left unchanged.
```

Reply:
The `is_stdin` check should be before the `p.is_dir()` to avoid an unnecessary disk read.